### PR TITLE
doc: pre-release doc updates

### PR DESCRIPTION
**Pull request by annwoj**
fixed links, pdk occurences.
other small changes

## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: main

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

## Description

JIRA ticket: 

## Self review

- [ ] There is no commented code.
- [ ] There are no TODO/FIXME comments without associated issue ticket.
- [ ] Commits are properly organized.
- [ ] Change has been tested.
- [ ] Tests were updated (if applicable).

### CODE REVIEW

File: doc/releases_and_migration/migration_guide_v280.rst [hunk 1]
```diff
@@ -116,20 +116,20 @@ See the following example of device characteristics:
...
   uart:~$ sid option -gsi
   [00:00:55.582,641] <inf> sid_cli: sid_option returned 0 (SID_ERROR_NONE); SIDEWALK_ID: BFFFFFABCD
 
 Fixing uplink buffer
 ====================
 
-Upon migration, you can remove the ``pending_message_list`` workaround, as the first uplink message after Link Up payload has been fixed in the v2.8.0 release. 
-For the issue details, see Amazon Sidewalk Known Issues.
+Upon migration, you can remove the ``pending_message_list`` workaround, as the first uplink message after Link Up payload has been fixed in the v2.8.0 release.
+For the issue details, see Amazon Sidewalk section of `Known issues for the nRF Connect SDK for the v2.8.0`_.
```
Comment (annwoj) [inline doc/releases_and_migration/migration_guide_v280.rst:135]: @ktaborowski has this item been added to known issues?

File: doc/samples/samples_list.rst [hunk 2]
```diff
@@ -11,22 +11,20 @@ The following table demonstrates the variants' configuration for the supported d
...
 |                                   | having activated CLI.                    | * `nRF5340 DK`_                | * `Semtech SX1262MB2CAS`_ (only for the configurations with LoRa or FSK)  | * Application partition size = 956 kB         |                           |
-|                                   |                                          | * `nRF54L15 PDK`_ `*`          |                                                                           |                                               |                           |
+|                                   |                                          | * `nRF54L15 DK`_               |                                                                           |                                               |                           |
 +-----------------------------------+------------------------------------------+--------------------------------+---------------------------------------------------------------------------+-----------------------------------------------+---------------------------+
 
-`*` The `nRF54L15 PDK`_ does not support LoRa and FSK configurations.
```
Comment (annwoj) [inline doc/samples/samples_list.rst:28]: @ktaborowski removed the note, let me know if this is still applicable

File: doc/compatibility_matrix.rst [hunk 3]
```diff
@@ -15,27 +15,41 @@ The following table demonstrates the compatibility between the nRF Connect SDK a
      - Amazon specification (Sidewalk MCU SDK version)
      - Supported platforms
      - Amazon qualification status
-   * - v1.14.3
-     - v2.3.0
+   * - ---
```
Comment (annwoj) [inline doc/compatibility_matrix.rst:18]: @ktaborowski @RobertGalatNordic could you check the compatibility table? And if there's anything that needs to be linked (the file that you've mentioned), could you add the suggestion as well?

File: doc/samples/samples_list.rst (hunk 2 above)
Comment (RobertGalatNordic) [reply doc/samples/samples_list.rst:28]: nrf54l15 dk supports all transports now.

Comment (annwoj) [reply doc/samples/samples_list.rst:28]: leaving as is then, thank you!

File: doc/releases_and_migration/migration_guide_v280.rst (hunk 1 above)
Comment (RobertGalatNordic) [reply doc/releases_and_migration/migration_guide_v280.rst:135]: It should be rather removed from known issues, as it is already fixed. I think that adding a note in migration guide is a good idea, as the temporary workaround has been done in the application layer, and users should know it is no longer necessary

File: doc/compatibility_matrix.rst (hunk 3 above)
Comment (RobertGalatNordic) [reply doc/compatibility_matrix.rst:18]: no, as far as I'm aware qualification of 2.7.0 is still ongoing so everything is in order.

**Review by RobertGalatNordic — APPROVED** [review]

File: doc/compatibility_matrix.rst (hunk 3 above)
Comment (annwoj) [reply doc/compatibility_matrix.rst]: files to be hosted on zoomin, tba

File: doc/releases_and_migration/migration_guide_v280.rst (hunk 1 above)
Comment (ktaborowski) [reply doc/releases_and_migration/migration_guide_v280.rst:135]: I dont see this error in known issues list for amazon sidewalk 2.7.0. It seems we forgot to add it in documentation. So, no reference in this case.

**Review by ktaborowski — APPROVED** [review]
lgtm

**Pull request merged**